pkg/rbac/rbac.go

```diff
@@ -0,0 +1,314 @@
+// Package rbac provides various Access Review API
+package rbac
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+
+	authorizationv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+)
+
+// ACLConfig holds the  access control configuration  needed to  perform an action
+// e.g. "get" permission on the  "managedclusters" resource is needed to "view" a managedcluster on the ACM Hub
+// The configuration includes ApiGroup, Resource type but not Version as rules in K8s ClusterRole
+// do not include/specify version.
+type ACLConfig struct {
+	// groupRes is the API Group and Resource information
+	groupRes schema.GroupResource
+	// verb is the action on the Resource
+	verb string
+}
+
+// MetricsACLConfig is an instance of ACLConfig and holds configuration for accessing observability
+// metrics gathered from ManagedClusters
+var MetricsACLConfig = ACLConfig{
+	groupRes: schema.GroupResource{
+		Group:    "cluster.open-cluster-management.io",
+		Resource: "managedclusters",
+	},
+	verb: "metrics/",
+}
+
+// AccessReviewer is the  API for custom finer-grained access control, it holds the
```

Comment:
Not sure if you meant this instead.
```suggestion
// AccessReviewer is the  API for custom fined-grained access control, it holds the
```